PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/RunSparkScalaLivyConsoleAction.kt

```diff
@@ -22,12 +22,32 @@
 
 package com.microsoft.azure.hdinsight.spark.console
 
-import com.microsoft.azure.hdinsight.common.logger.ILogger
+import com.intellij.execution.RunManagerEx
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.microsoft.azure.hdinsight.spark.run.configuration.ArisSparkConfiguration
+import com.microsoft.azure.hdinsight.spark.run.configuration.CosmosSparkRunConfiguration
+import com.microsoft.azure.hdinsight.spark.run.configuration.LivySparkBatchJobRunConfiguration
 import org.jetbrains.plugins.scala.console.ScalaConsoleRunConfigurationFactory
 
-class RunSparkScalaLivyConsoleAction : RunSparkScalaConsoleAction(), ILogger {
+class RunSparkScalaLivyConsoleAction : RunSparkScalaConsoleAction() {
     override val consoleRunConfigurationFactory: ScalaConsoleRunConfigurationFactory
         get() = SparkScalaLivyConsoleConfigurationType().confFactory()
 
     override fun getNewSettingName(): String = "Spark Livy Interactive Session Console(Scala)"
+
+    override fun update(event: AnActionEvent) {
+        super.update(event)
```

Comment:
It might be better to disable it firstly at beginning:
`event.presentation.isEnabled = false`

Otherwise, the following return when project is null will cause the presentation is enabled by default.